### PR TITLE
Prepare 4.0.0-alpha1 release

### PR DESCRIPTION
**Pull request by core23**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,23 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [4.0.0-alpha-1](https://github.com/sonata-project/SonataClassificationBundle/compare/3.17.0...4.0.0-alpha-1) - 2021-11-01
```
Comment (ChristianGripp) [inline CHANGELOG.md]: Predated this to tomorrow

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,23 @@
...
+### Added
+- [[#754](https://github.com/sonata-project/SonataClassificationBundle/pull/754)] Add type hints to manager methods ([@core23](https://github.com/core23))
```
Comment (phansys) [inline CHANGELOG.md]:
```suggestion
- [[#754](https://github.com/sonata-project/SonataClassificationBundle/pull/754)] Type hints to manager methods ([@core23](https://github.com/core23))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,23 @@
...
+- [[#752](https://github.com/sonata-project/SonataClassificationBundle/pull/752)] Make blocks editable ([@core23](https://github.com/core23))
+- [[#753](https://github.com/sonata-project/SonataClassificationBundle/pull/753)] Added `ContextAwareInterface` ([@core23](https://github.com/core23))
```
Comment (phansys) [inline CHANGELOG.md]:
```suggestion
- [[#753](https://github.com/sonata-project/SonataClassificationBundle/pull/753)] `ContextAwareInterface` ([@core23](https://github.com/core23))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,23 @@
...
+
+### Changed
+- [[#745](https://github.com/sonata-project/SonataClassificationBundle/pull/745)] Added type hints to methods and properties ([@core23](https://github.com/core23))
+- [[#745](https://github.com/sonata-project/SonataClassificationBundle/pull/745)] Reduce visibility of config methods ([@core23](https://github.com/core23))
+
+### Fixed
+- [[#757](https://github.com/sonata-project/SonataClassificationBundle/pull/757)] Removed call to unknown `FormMapper::getAdmin` method ([@core23](https://github.com/core23))
+
+### Removed
+- [[#745](https://github.com/sonata-project/SonataClassificationBundle/pull/745)] Removed `Sonata\ClassificationBundle\Form\FormHelper` ([@core23](https://github.com/core23))
```
Comment (phansys) [inline CHANGELOG.md]:
```suggestion
- [[#745](https://github.com/sonata-project/SonataClassificationBundle/pull/745)] `Sonata\ClassificationBundle\Form\FormHelper` ([@core23](https://github.com/core23))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,23 @@
...
+- [[#744](https://github.com/sonata-project/SonataClassificationBundle/pull/744)] Dropped support for PHP 7.3 ([@core23](https://github.com/core23))
```
Comment (phansys) [inline CHANGELOG.md]:
```suggestion
- [[#744](https://github.com/sonata-project/SonataClassificationBundle/pull/744)] Support for PHP 7.3 ([@core23](https://github.com/core23))
```

**Review by phansys — DISMISSED** [review]

**Review by phansys — DISMISSED** [review]

**Comment by jordisala1991** [comment]
Can you use alpha1 like in the media-bundle?

**Comment by core23** [comment]
We have three different version names at the moment (created by three different people):

https://github.com/sonata-project/SonataDoctrineORMAdminBundle/releases/tag/4.0.0-alpha-1
https://github.com/sonata-project/SonataSeoBundle/releases/tag/3.0.0-alpha.1
https://github.com/sonata-project/SonataMediaBundle/releases/tag/4.0.0-alpha1

Not sure what is the *right* one.

**Comment by jordisala1991** [comment]
> We have three different version names at the moment (created by three different people):
> 
> 
> 
> https://github.com/sonata-project/SonataDoctrineORMAdminBundle/releases/tag/4.0.0-alpha-1
> 
> https://github.com/sonata-project/SonataSeoBundle/releases/tag/3.0.0-alpha.1
> 
> https://github.com/sonata-project/SonataMediaBundle/releases/tag/4.0.0-alpha1
> 
> 
> 
> Not sure what is the *right* one.

https://getcomposer.org/doc/articles/versions.md#vcs-tags-and-branches

alpha1 from composer pov

**Comment by core23** [comment]
Changed this to `alpha1`

**Comment by VincentLanglet** [comment]
I'll try to bump the static analysis but this can be done between the alpha and the stable one

**Review by VincentLanglet — DISMISSED** [review]

**Comment by VincentLanglet** [comment]
See https://github.com/sonata-project/SonataClassificationBundle/pull/759

**Comment by core23** [comment]
> See #759

Is this a blocking PR? There are only internal changes and no API changes.

**Comment by VincentLanglet** [comment]
> > See #759
> 
> Is this a blocking PR? There are only internal changes and no API changes.

Not blocking for the alpha. It will for the RC (since we deprecate some stuff)

**Review by jordisala1991 — CHANGES_REQUESTED** [review]
Please include https://github.com/sonata-project/SonataClassificationBundle/pull/761

**Comment by VincentLanglet** [comment]
phpstan/psalm PR is ready if you want @core23 https://github.com/sonata-project/SonataClassificationBundle/pull/759

**Comment by core23** [comment]
Please review @sonata-project/contributors. I'll plan to release this tomorrow

**Review by jordisala1991 — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**